tests/npm-app/src/endpoints/crypto.ts

```diff
@@ -27,24 +27,47 @@ export function generateAesKey(
 
 type Base64 = string;
 
-interface WrapKeyRsaOaepRequest {
+interface WrapAlgoParams {
+  name: string;
+}
+
+interface RsaOaepParams extends WrapAlgoParams {
+  label?: Base64;
+}
+
+interface RsaOaepAesParams extends WrapAlgoParams {
+  aes_key_size: number; // in bits
+  label?: Base64;
+}
+
+interface WrapKeyRequest {
   key: Base64; // typically an AES key
-  wrappingKey: string; // PEM-encoded RSA public key
-  label?: string;
+  wrappingKey: Base64; // base64 encoding of PEM-encoded RSA public key or AES key bytes
+  parameters: WrapAlgoParams; // Wrapping algorithm parameters
 }
 
-export function wrapKeyRsaOaep(
-  request: ccf.Request<WrapKeyRsaOaepRequest>
+export function wrapKey(
+  request: ccf.Request<WrapKeyRequest>
 ): ccf.Response<ArrayBuffer> {
   const r = request.body.json();
   const key = b64ToBuf(r.key);
-  const wrappingKey = publicPemToDer(r.wrappingKey);
-  const label = r.label ? ccf.ccf.strToBuf(r.label) : undefined;
-  const wrappedKey = ccf.ccf.wrapKey(key, wrappingKey, {
-    name: "RSA-OAEP",
-    label: label,
-  });
-  return { body: wrappedKey };
+  const wrappingKey = b64ToBuf(r.wrappingKey);
+  if (r.parameters.name == "RSA-OAEP") {
+    const p = r.parameters as RsaOaepParams;
+    const l = p.label ? b64ToBuf(p.label) : undefined;
+    const new_p = { name: p.name, label: l };
+    const wrappedKey = ccf.ccf.wrapKey(key, wrappingKey, new_p);
+    return { body: wrappedKey };
+  } else if (r.parameters.name == "RSA-OAEP-AES-KWP") {
+    const p = r.parameters as RsaOaepAesParams;
+    const l = p.label ? b64ToBuf(p.label) : undefined;
+    const new_p = { name: p.name, aes_key_size: p.aes_key_size, label: l };
+    const wrappedKey = ccf.ccf.wrapKey(key, wrappingKey, new_p);
+    return { body: wrappedKey };
```

Comment:
@letmaik I hate the base64 decoding going on here, especially for the labels. Is there a way to avoid that or make the infrastructure do it without having to drill into the `WrapAlgoParams`? Maybe just move the base64 decoding into C++ so we don't have to replicate the logic here?

Reply:
Whether JS apps use Base64 for the label is up to them, we shouldn't prescribe that. I think it's ok as you have it.